rs/nns/governance/src/governance/test_data.rs

```diff
@@ -118,7 +118,8 @@ lazy_static! {
             }),
         }),
         dapp_canisters: vec![pb::Canister {
-            id: Some(CanisterId::from_u64(1000).get())
+            // must match `SPECIFIED_CANISTER_ID` defined in `ic-nns-test-utils`
+            id: Some(CanisterId::from_u64(0x2100000).get())
```

Comment:
Why does this have to match that value?  That seems somewhat accidental to me.

Reply:
It is not a hard requirement to match this exact constant, but I thought it'd be better understandable than the actual condition: the canister ID must belong to the [extra canister range](https://github.com/dfinity/ic/blob/d7fcb9aa41cf3aa4d9991db8de1ee043aedb6f4b/rs/nns/test_utils/src/state_test_helpers.rs#L103) set up in `state_machine_builder_for_nns_tests`.

Reply:
Please see: https://github.com/dfinity/ic/pull/5269/commits/2799a7c62727dd132f00b84207ea46d385c30b3a

Reply:
And https://github.com/dfinity/ic/pull/5269/commits/51cdd51b61328b03d84bc3b5f6eaa0c5e87db874 to avoid dead code.